quinn/examples/single_socket.rs

```diff
@@ -0,0 +1,144 @@
+//! QUIC does not care about the local endpoints - connection is not associated with source IP and
```

Comment:
I think this could be potentially confusing. We have the `Endpoint` type which should correspond to exactly one UDP address (IP + port), otherwise things will go wrong. So QUIC does care about local endpoints -- the key point you're trying to make, I think, is that we can run an arbitrary number of connections over a single endpoint. (Also this documentation does not discuss that you can have "server" and "client" connections on the same endpoint, but that might be worth pointing out as well?)

Reply:
Removed the misleading info, but I'm not sure for 
> Also this documentation does not discuss that you can have "server" and "client" connections on the same endpoint

You mean pointing out that "server" and "client" both can have the same endpoint (`source_ip:port`)? If so, maybe a separate example for this case?

Although, I don't know the real world use case for this and hence such example seems synthetic to me :/

Reply:
For example in a peer-to-peer scenario, a particular process might be acting as the "server" for one peer, while it acts as the "client" for another peer.

Reply:
good point, will put this in the docs at the very least :)